delfin/leader_election/factory.py

```diff
@@ -0,0 +1,45 @@
+# Copyright 2021 The SODA Authors.
+# Copyright 2010 United States Government as represented by the
+# Administrator of the National Aeronautics and Space Administration.
+# Copyright 2011 Justin Santa Barbara
+# All Rights Reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License"); you may
+#    not use this file except in compliance with the License. You may obtain
+#    a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+
+from delfin.leader_election.tooz.callback import ToozLeaderElectionCallback
+from delfin.leader_election.tooz.leader_elector import Elector
+from delfin.task_manager.scheduler.schedule_manager import SchedulerManager
+
+LEADER_ELECTION_KEY = "delfin-performance-metric-collection"
+
+
+class LeaderElectionFactory:
+
+    @staticmethod
+    def construct_elector(plugin, leader_key=None):
+        # maintain a unique key for metric collection leader election
+        leader_election_key = LEADER_ELECTION_KEY
+        if leader_key:
+            leader_election_key = leader_key
+
+        scheduler_mgr = SchedulerManager()
+
+        if plugin == "tooz":
```

Comment:
plugin name can be a configurable name, so that it is easy to switch from one plugin type to other in future  based on configuration..

Reply:
Its already a configurable parameter with name [leader_election_plugin](https://github.com/sodafoundation/delfin/pull/589/files#diff-f264922ca8b4b4a105e21667c9cbadb943b1fc417ef4c2a2f993624782426c69R72).